src/auth/middleware.js

```diff
@@ -1,47 +1,66 @@
 'use strict';
-
+/**
+ * Authentication Middleware Module - Performs basic access authentication on usernames and passwords attached to request object headers
+ * @module auth/middleware
+ */
 const User = require('./users-model.js');
 
 module.exports = (req, res, next) => {
 
   try {
 
-    let [authType, encodedString] = req.headers.authorization.split(/\s+/);
+    let [authType, authString] = req.headers.authorization.split(/\s+/);
+
+    console.log('authType', authType);
 
     // BASIC Auth  ... Authorization:Basic ZnJlZDpzYW1wbGU=
 
-    switch(authType.toLowerCase()) {
+    switch (authType.toLowerCase()) {
       case 'basic':
-        return _authBasic(encodedString);
+        return _authBasic(authString);
       default:
         return _authError();
     }
 
-  } catch(e) {
+  } catch (e) {
     return _authError();
   }
-
-  function _authBasic() {
-    let base64Buffer = Buffer.from(authString,'base64'); // <Buffer 01 02...>
+  /** @function
+   * @name _authBasic Parses authentication header data and passes into a User model method to query the database for matching username/password.
+   * @param  {string} authString string attached to the request authentication header
+   */
+  function _authBasic(authString) {
```

Comment:
There were 3 bugs in here and 2 in auth/users-model.js and it looks like you got them all, awesome job.